src/main/java/org/victorrobotics/frc/dtlib/DTRobot.java

```diff
@@ -1,28 +1,144 @@
 package org.victorrobotics.frc.dtlib;
 
-import java.util.HashSet;
-import java.util.Set;
+import org.victorrobotics.frc.dtlib.command.test.DTSelfTestCommand;
+import org.victorrobotics.frc.dtlib.command.util.DTPrintCommand;
 
+import java.util.ArrayList;
+import java.util.List;
+
+import edu.wpi.first.wpilibj.Compressor;
 import edu.wpi.first.wpilibj.TimedRobot;
+import edu.wpi.first.wpilibj2.command.Command;
 import edu.wpi.first.wpilibj2.command.CommandScheduler;
 
 public abstract class DTRobot extends TimedRobot {
-    private final Set<DTSubsystem> subsystems;
+    private final List<DTSubsystem> subsystems;
+    private Compressor              pneumaticsCompressor;
+    private boolean                 compressorEnabled;
+
+    private Command autonomousCommand;
 
     protected DTRobot() {
         super(0.02);
-        subsystems = new HashSet<>();
+        subsystems = new ArrayList<>();
+        pneumaticsCompressor = null;
+        compressorEnabled = true;
     }
 
-    @Override
-    public void robotPeriodic() {
-        
+    protected abstract void onBootUp();
+
+    protected abstract void bindCommands();
+
+    protected abstract Command getAutonomousCommand();
+
+    public final DTSelfTestCommand getSelfTestCommand() {
+        return new DTSelfTestCommand(subsystems.toArray(DTSubsystem[]::new));
     }
 
     protected final void registerSubsystem(DTSubsystem subsystem) {
         subsystems.add(subsystem);
     }
 
+    protected final void setCompressor(Compressor compressor) {
```

Comment:
Should we have a method to create a standard compressor object instead?

Reply:
Perhaps. Probably two, since there are two types of compressors that can be created (or an enum, since I believe one already exists)